File: app/src/main/java/hr/ml/moviesdiscover/rest/model/DiscoveredMoviesFromRetrofit.java
package hr.ml.moviesdiscover.rest.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class DiscoveredMoviesFromRetrofit {

    @SerializedName("page")
    @Expose
    private Integer page;
    @SerializedName("results")
    @Expose
    private List<MovieFromRetrofit> results;
    @SerializedName("total_pages")
    @Expose
    private Integer totalPages;
    @SerializedName("total_results")
    @Expose
    private Integer totalResults;

    public Integer getPage() {
        return page;
    }

    public void setPage(Integer page) {
        this.page = page;
    }

    public List<MovieFromRetrofit> getResults() {
        return results;
    }

    public void setResults(List<MovieFromRetrofit> results) {
        this.results = results;
    }

    public Integer getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(Integer totalPages) {
        this.totalPages = totalPages;
    }

    public Integer getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(Integer totalResults) {
        this.totalResults = totalResults;
    }
}

File: app/src/main/kotlin/hr/ml/moviesdiscover/rest/model/DiscoveredMoviesFromRetrofit.kt
package hr.ml.moviesdiscover.rest.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class DiscoveredMoviesFromRetrofit(

        @SerializedName("page")
        @Expose
        val page: Int?,

        @SerializedName("results")
        @Expose
        val results: List<MovieFromRetrofit>?,

        @SerializedName("total_pages")
        @Expose
        val totalPages: Int?,

        @SerializedName("total_results")
        @Expose
        val totalResults: Int?

)
